lib/util.js

```diff
@@ -147,11 +150,61 @@ function styleText(format, text, { validateStream = true, stream = process.stdou
       validateOneOf(key, 'format', ObjectKeys(inspect.colors));
     }
     if (skipColorize) continue;
-    left += escapeStyleCode(formatCodes[0]);
-    right = `${escapeStyleCode(formatCodes[1])}${right}`;
+    ArrayPrototypePush(codes, formatCodes);
   }
 
-  return skipColorize ? text : `${left}${text}${right}`;
+  if (skipColorize) {
+    return text;
+  }
+
+  // Build opening codes
+  let openCodes = '';
+  for (let i = 0; i < codes.length; i++) {
+    openCodes += escapeStyleCode(codes[i][0]);
+  }
+
+  // Process the text to handle nested styles
+  let processedText;
+  if (codes.length > 0) {
+    processedText = ArrayPrototypeReduce(
+      codes,
+      (text, code) => RegExpPrototypeSymbolReplace(
+        new RegExp(`\\u001b\\[${code[1]}m`, 'g'),
+        text,
+        (match, offset) => {
+          // Check if there's more content after this reset
+          if (offset + match.length < text.length) {
+            const opening = escapeStyleCode(code[0]);
+            // Only reapply the style if there's not already an opening code
+            if (StringPrototypeIncludes(text, opening)) {
```

Comment:
is this an optimisation?

think of this example:

```ts
red(
  'red' +
  blue(
    'blue' +
    red('red') +
    'blue'
  ) +
  'red'
);
```

```
44m
  red
  32m
    blue
    44m red 32m
    blue
  0m <- this would've been 44m, but `text` includes `44m` so L180 will be hit (i think?)
  red
0m

Reply:
fixed this case, thanks